azurelinuxagent/common/protocol/wire.py

```diff
@@ -844,14 +852,9 @@ def _call_hostplugin_with_container_check(self, host_func):
         """
         Calls host_func on host channel and accounts for stale resource (ResourceGoneError or InvalidContainerError).
         If stale, it refreshes the goal state and retries host_func.
-        This method can throw, so the callers need to handle that.
         """
         try:
-            ret = host_func()
-            if ret in (None, False):
```

Comment:
This code was trying to handle the different error values (False, None, exceptions) of the different downloads. Now all functions use exceptions so this is no longer needed